src/Solnet.Programs/NameServiceProgram.cs

```diff
@@ -228,15 +226,15 @@ public static PublicKey DeriveNameAccountKey(byte[] hashedName, PublicKey nameCl
         /// <param name="lamports">The number of lamports for rent exemption.</param>
         /// <param name="space">The space for the account.</param>
         /// <returns>The transaction instruction data.</returns>
-        private static byte[] EncodeCreateNameRegistryData(byte[] hashedName, ulong lamports, int space)
+        private static byte[] EncodeCreateNameRegistryData(ReadOnlySpan<byte> hashedName, ulong lamports, uint space)
         {
             byte[] methodBuffer = new byte[49];
 
-            methodBuffer[0] = (byte)NameServiceInstructions.Create;
-            Utils.Uint32ToByteArrayLe((ulong) hashedName.Length, methodBuffer, 1);
-            Array.Copy(hashedName, 0, methodBuffer, 5, hashedName.Length);
-            Utils.Int64ToByteArrayLe(lamports, methodBuffer, 37);
-            Utils.Uint32ToByteArrayLe((ulong) space, methodBuffer, 45);
+            methodBuffer.WriteU8((byte)NameServiceInstructions.Create, 0);
+            methodBuffer.WriteU32((uint) hashedName.Length, 1);
+            methodBuffer.WriteSpan(hashedName, 5);
+            methodBuffer.WriteU64(lamports, 37);

```

Comment:
I have to say that its so much better to look at and reason with after these changes